pkg/services/preference/prefapi/api.go

```diff
@@ -96,6 +97,21 @@ func GetPreferencesFor(ctx context.Context,
 			dto.Language = &preference.JSONData.Language
 		}
 
+		if preference.JSONData.Navbar.SavedItems != nil {
```

Comment:
Does this mean that we'll need to send all the preferences data every time the user pins a nav item? What about removing an item from the saved items?

Reply:
We can use the `patch` endpoint for all of these cases, but yes if we use `update` we always need to send _all_ the preferences (this is what happens when you press save on the Preferences page).